superset/db_engine_specs/hive.py

```diff
@@ -525,3 +526,13 @@ def get_function_names(cls, database: "Database") -> List[str]:
         :return: A list of function names useable in the database
         """
         return database.get_df("SHOW FUNCTIONS")["tab_name"].tolist()
+
+    @classmethod
+    def is_readonly(cls, parsed_query: ParsedQuery) -> bool:
+        """Pessimistic readonly, 100% sure statement won't mutate anything"""
+        return (
+            parsed_query.is_select()
+            or parsed_query.is_explain()
+            or parsed_query.is_set()
+            or parsed_query.is_show()
```

Comment:
To avoid duplication of logic, we should probably do `super().is_readonly() or ...` here.

Reply:
done